### DISCUSSION TITLE
Namron Awning Controller - Cover commands not working

### DISCUSSION THREAD

**Discussion by Swoop86** [zwave-js/zwave-js#7089] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
After including the device, it shows with the "outbound bottom" buttons (up, stop and down). When using the zip screen, the device reports voltage and amperes correctly, but when trying to control the zip screen up, down or a select position an "Failed to call service cover/open_cover. unknown error" occurs.

What did you expect to happen instead?
Zip screen should go up, down or be positioned at the given setting.

N/A

Anything else we should know?
[zwave_js-65b5997be8aa9486f781a436bd9f2326-Node 134-92a842cc87918f4c8f4005eedb92a058.json](https://github.com/user-attachments/files/16511254/zwave_js-65b5997be8aa9486f781a436bd9f2326-Node.134-92a842cc87918f4c8f4005eedb92a058.json)
[home-assistant_zwave_js_2024-08-06T13-27-23.331Z.log](https://github.com/user-attachments/files/16511257/home-assistant_zwave_js_2024-08-06T13-27-23.331Z.log)

### Software versions

Driver (node-zwave-js): 12.11.2
Home Assistant Z-Wave Integration: 0.6.2

### Device information

Manufacturer: Namron
Model name: Z-Wave markisekontroller 2A
Node ID: 134

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[home-assistant_zwave_js_2024-08-06T14-00-25.130Z.log](https://github.com/user-attachments/files/16511379/home-assistant_zwave_js_2024-08-06T14-00-25.130Z.log)
[zwave_js-65b5997be8aa9486f781a436bd9f2326-Node 134-92a842cc87918f4c8f4005eedb92a058.json](https://github.com/user-attachments/files/16511380/zwave_js-65b5997be8aa9486f781a436bd9f2326-Node.134-92a842cc87918f4c8f4005eedb92a058.json)

**Comment by Swoop86** [comment]
The device gets a new light entity when calibrating, when included with S2, this entity does not work as z-wave js logs says entity does not support Command Class Multilevel Switch. 
When included without security, this light entity can control the blinds down and up.
[ZWaveError.txt](https://github.com/user-attachments/files/16551272/ZWaveError.txt)
<img width="1040" alt="bilde" src="https://github.com/user-attachments/assets/7625b8d5-16b2-4f8b-96ce-5a6599790758">

**Reply by AlCalzone** [reply]
That one shouldn't exist, see https://github.com/zwave-js/node-zwave-js/issues/6318

**Comment by AlCalzone** [comment]
Without looking at the logs, this is probably https://github.com/zwave-js/certification-backlog/issues/32 or https://github.com/zwave-js/certification-backlog/issues/30, which are both bugs in Home Assistant.

Will double check to be sure.

**Reply by AlCalzone** [reply, marked as answer]
Confirmed, this is https://github.com/zwave-js/certification-backlog/issues/32, which is actually a bug in the Z-Wave integration in HA.

**Comment by Swoop86** [comment]
Following up on this with data from debug logs + a full diagnostics dump on 
Namron 4512748 (manufacturerId 0x0438, productType 0x0004, productId 0xD00D, 
firmware 2.4). As the last Z-wave version working is 0.29.1, anything higher breaks how the device works.

Confirmed: the device advertises both Window Covering CC (0x6A) and Multilevel 
Switch CC (0x26) on the root device and both endpoints (1 and 2), matching what's 
already been reported here.

New finding: when WindowCoveringCCSupportedGet is queried, the device only 
reports "(no position)" parameters - "Outbound Bottom (no position)" on endpoint 1, 
"Horizontal Slats Angle (no position)" on endpoint 2. So even once Window Covering 
CC is correctly exposed per spec, there's no percentage-based position support in 
the device's implementation of that CC at all - only a binary open/closed toggle. 
This means the cover entity that gets created isn't just non-functional due to a 
mapping issue, the underlying firmware genuinely doesn't implement position 
control under that CC.

Questions:
1. Is a local compat override (commandClasses.remove for 0x6A across all 
   endpoints) the right interim approach to keep Multilevel Switch CC exposed and 
   preserve current (working) light/cover behavior until firmware is fixed? I've 
   set this up locally and wanted to confirm before relying on it long-term.
2. I'm willing to submit a device config PR for this device. Given the firmware 
   limitation above, would the project prefer a config that includes the compat 
   removal (to keep it working as-is), or would that be discouraged since it 
   suppresses spec-mandated behavior? Happy to follow whatever convention makes 
   sense here.

I've also reported this to Namron directly with a reference to SDS13781 and the 
"(no position)" finding, so there may be a firmware fix coming, but no ETA yet.


[zwave_js-65b5997be8aa9486f781a436bd9f2326-Node 159-55f924d3772dc9ca372b4c6e0b5a3fa1.json](https://github.com/user-attachments/files/28993715/zwave_js-65b5997be8aa9486f781a436bd9f2326-Node.159-55f924d3772dc9ca372b4c6e0b5a3fa1.json)

[namron.rtf](https://github.com/user-attachments/files/28993716/namron.rtf)

**Reply by AlCalzone** [reply]
If the device is fully operational with Multilevel Switch CC, and Window Covering CC support is broken, suppressing `Window Covering CC` through a compat flag is the correct approach IMO.

**Reply by Swoop86** [reply]
Update: Compat override works at the driver level, but stale Window Covering entities persist in Home Assistant

Following up after testing the removeCCs compat override on Z-Wave JS 1.4.0 (HA 2026.5.4):

What works:
- User-provided device config loaded confirmed in logs
- Window Covering CC (0x6A) interview is completely skipped on all endpoints
- Multilevel Switch CC (0x26) is interviewed successfully on all endpoints with currentValue (0–99)
- Interview completes: "The node is ready to be used"

What doesn't work:

A- fter re-interview with the compat override, Window Covering cover entities still appear in HA as active entities ("Outbound Bottom", "Horizontal Slats Angle")
- These entities cannot be deleted from HA's entity registry — the delete button is greyed out because the integration is still "providing" them
- The stale cover entities appear to block the correct Multilevel Switch entities from being created/used properly

The removeCCs compat flag removes Window Covering CC from the interview, but the driver still does a version query for the CC during the interview's "Version" stage (since it's in the device's NIF). This version query appears to be enough for the Z-Wave JS server to keep advertising Window Covering entities to HA, even though no values exist for them.

In the Z-Wave JS UI diagnostics dump, the values section shows no Window Covering CC values (confirming removeCCs works at the value level), but the entities section still lists Window Covering cover entities with primary_value: null.

Questions:
1) Should removeCCs also suppress the version query for the removed CC? Currently the driver queries the CC version even though the CC is marked for removal, which seems to cause the stale entity issue.
2) Is there a way to force HA to discard the old Window Covering entities without removing and re-including the device? (This affects 8 sun screens, so re-inclusion is a significant effort.)

**Reply by AlCalzone** [reply]
> Currently the driver queries the CC version even though the CC is marked for removal, which seems to cause the stale entity issue

It _should_ be determined by the support flag, not the version. But there might be a bug.

> Is there a way to force HA to discard the old Window Covering entities without removing and re-including the device?

Just a re-interview would be the way, but that requires Z-Wave JS not to expose the entities in the first place.

Can you share the following please:
- A Z-Wave debug log of the interview with the compat flag enabled
- A Z-Wave debug log of you controlling the device, ideally outside of HA
- A driver dump of the device (Apps -> Z-Wave JS -> Open Web UI -> Click the node -> Advanced -> <img width="226" height="145" alt="image" src="https://github.com/user-attachments/assets/80d806d1-7f27-4374-970d-192e1912bd9d" />

**Reply by Swoop86** [reply]
Sorry for the slow progress here, but things came up and I've been unable to look at this. 
Will hopefully be able to provide you with the logs during this week.

**Comment by Swoop86** [comment]
Getting a 2026-06-28 12:01:35.609 CNTRLR   [Node 159] User-provided device config loaded, when z-wave js is booting up.

I don't have any switches connected to the z-wave device, so only way to control them is via z-wave (:)), but i tried a re-calibration as that has been necessary to get the light entity up and running.
Tried Z-wave JS 1.5.0, but still no luck, reverting back to 0.29.1 to get them working again. What kind of sucks now, is that upgrading HA forces Z-wave to be upgraded as well, so I'm kind of stuck at the current version of HA as well. Hope you have some ideas.

[re-calibration.rtf](https://github.com/user-attachments/files/29434951/re-calibration.rtf)
[node_159_dump-2.json](https://github.com/user-attachments/files/29434952/node_159_dump-2.json)
[re-interview.rtf](https://github.com/user-attachments/files/29434953/re-interview.rtf)

**Reply by AlCalzone** [reply]
Please show me your `compat` section in the config file too.

**Reply by Swoop86** [reply]
[namron_4512748.json](https://github.com/user-attachments/files/29458923/namron_4512748.json)

**Reply by AlCalzone** [reply]
I'm not sure what you are doing, but that's not a valid JSON file. This looks like RTF.

Adding Multilevel Switch CC to all endpoints should not be needed - removing Window Covering should be enough.

**Reply by Swoop86** [reply]
Didn't have a proper way of exporting it, this is how it looks like in HA:
<img width="474" height="891" alt="image" src="https://github.com/user-attachments/assets/1da14323-d84d-46b7-86f1-c1215a467924" />

Tried this as well, but with same results:
<img width="467" height="522" alt="image" src="https://github.com/user-attachments/assets/1212b837-9d38-4677-85bc-acf813bc4404" />

**Reply by AlCalzone** [reply]
That looks better. Can you use the latter one, but without the Basic Set line?
Then restart the Z-Wave JS app, re-interview the device.

Share a Z-Wave log of the re-interview, the HA diagnostics of the device after the interview.

Also include a log of you interacting with the device, so I see again what it sends.

**Reply by Swoop86** [reply]
Here you go
[namron outbound ctrl.txt](https://github.com/user-attachments/files/29463112/namron.outbound.ctrl.txt)
[namron re-calibration.txt](https://github.com/user-attachments/files/29463114/namron.re-calibration.txt)
[namron re-interview.txt](https://github.com/user-attachments/files/29463115/namron.re-interview.txt)
[node_159_dump-3.json](https://github.com/user-attachments/files/29463120/node_159_dump-3.json)

**Reply by AlCalzone** [reply]
Thanks, something is definitely wrong there. I'll look into it.

**Reply by Swoop86** [reply]
Thanks, appreciate it! Let me know if there is anything more needed from my side.

**Reply by AlCalzone** [reply]
Found something in the interview log:
> 2026-06-29 13:08:19.273 CNTRLR   [Node 159] No device config found

Looks like your file isn't being picked up after all.

Where exactly did you put it? Did you restart the Z-Wave JS UI app afterwards?

**Reply by Swoop86** [reply]
Hm. Weird, might have forgotten to restart z-wave js this time. I recall doing this a few times before posting here last time, and even if it picked up the config the results were the same with that the screens were uncontrollable via HA. I'll give it another go to get the logs out.
It's located in the /config directory.

**Reply by Swoop86** [reply]
2026-06-29 14:27:20.184 CNTRLR   [Node 159] User-provided device config loaded

<img width="340" height="530" alt="image" src="https://github.com/user-attachments/assets/46269717-d88b-45e8-8fc6-0a10ed9a227c" />

[namron re-interview.txt](https://github.com/user-attachments/files/29465399/namron.re-interview.txt)

**Reply by AlCalzone** [reply]
The interview looks different now - I don't see Window Covering CC get interviewed now.
Try the interaction again and see how it behaves?

**Reply by Swoop86** [reply]
Well I have good news and bad news.
Good news:
After re-calibrating, I can now control it, not quite sure why I haven't been able to control them before. I will leave it as is and test with a few more screens to see how they behave.

Bad news:
Settings are inverted, when the screen is closed, it is shown as "open", when open it is showed as "closed". Any idea how to invert this?
Compared to when it was a "light bulb", there 100% on was closed and 0% was open.

<img width="502" height="589" alt="image" src="https://github.com/user-attachments/assets/8ca8f881-0b6e-4dd6-a024-5fab8bc23b58" />

Appreciate your time with this regardless, as I've felt I've been stuck for a few weeks trying this without any luck. Although I have done exactly the same as I've done today...

**Reply by AlCalzone** [reply]
Is this now with Window Covering CC disabled or not?

**Reply by Swoop86** [reply]
It's running with the following custom config:
<img width="502" height="589" alt="image" src="https://github.com/user-attachments/assets/8a385405-b5d0-46aa-b90c-7bee7e03c491" />

Following controls available after re-interview:
<img width="333" height="287" alt="image" src="https://github.com/user-attachments/assets/caf94ee1-499a-4781-95c7-e9319cfc9d4c" />

**Reply by AlCalzone** [reply]
So it is with Multilevel Switch CC, not Window Covering.

It looks like your device is another whitelabel of this device:
https://devices.zwave-js.io/?jumpTo=0x0330:0x0004:0xd00d:2.4.5

That one looks like it has a parameter that could influence how it interprets the values:
<img width="1098" height="310" alt="image" src="https://github.com/user-attachments/assets/200c0241-cfb7-4a39-9110-531c69dab075" />

Maybe you can try setting that parameter manually and see if this corrects how it reports closed vs. open?

In any case I would consider the current reporting to be wrong. The Z-Wave specs are pretty clear about how the values are meant to be interpreted. 100% / Open is always supposed to be the "most light" setting.

**Reply by Swoop86** [reply]
I looked into that setting as well, its default to 2, have tried setting it to 2 manually in the past as well but it had no effect.

Interestingly enough, after all the "work", getting the value (which shows it set as 2), then setting the value to 2, doing a re-interview seems to reset the position.

Will see how it works over the next few days, but thanks again for all the input here, will continue to set the value manually on the rest of the devices and do a re-interview. Will give feedback here if something isn't right.

Again, thanks!

**Reply by Swoop86** [reply]
Small update, the position reverts once the cover is used. So after re-interviewing it is listed as "closed" when the position is closed. Pressing the button to open, updates the status to "open" (but it's still closed). Then it's back to how it was (open is closed, closed is open). At least I can operate the screens now, but would be nice to know if this is solvable by other ways than making templates.

**Reply by AlCalzone** [reply]
Might be a dumb question but are you sure it is wired correctly?

**Reply by Swoop86** [reply]
Not dumb at all, I got it wired by my electrician, but might be a possibility that he's switched the wires...although a bit weird that when in "lightbulb" mode it reports and works correctly, I would expect the same behaviour when in cover mode, but being unfamiliar with this I have no idea. 
Might actually just be simpler to flip the wiring...

**Reply by Swoop86** [reply]
Not quite related to z-wave, but how would the screen react if the wires are switched?
If they are open, reporting they are closed, then the wire gets switched. then the status is technically correct, but wouldn't that just cause the screen to attempt to go further down again (although it is in a locked position, so the stop setting at the screen itself is stopping it)?

**Reply by AlCalzone** [reply]
Not entirely sure how this particular device works. I had some roller shutters before which were measuring the motor load to detect when the end point was reached. 

If I had wired them the opposite way, they would function completely inverse:
- open would close them, setting to 100% would also close them and they would report 100% when closed
- close would open them, setting to 0% would also open them and they would report 0% when opened

**Reply by Swoop86** [reply]
Works better with the wires inverted...
